python/triton/language/semantic.py

```diff
@@ -230,7 +230,15 @@ def add(input: tl.tensor | numbers.Number, other: tl.tensor | numbers.Number, sa
         input_scalar_ty = input.type.scalar
         other_scalar_ty = other.type.scalar
     if input_scalar_ty.is_ptr():
-        return tl.tensor(builder.create_addptr(input.handle, other.handle), input.type)
+        other_handle = other.handle
+        if other.dtype.is_int_unsigned() and other.dtype.int_bitwidth < 64:
+            # We want to zext when the offset is smaller than pointer size (64bit)
```

Comment:
NIT: Comments should explain why, not what
```suggestion
            # addptr treats offset as signed. Zero-extend unsigned offsets to ensure they're positive
```